tests/BenchmarkDotNet.Tests/Exporters/XmlSerializerTests.cs

```diff
@@ -0,0 +1,245 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using BenchmarkDotNet.Exporters.Xml;
+using Xunit;
+
+namespace BenchmarkDotNet.Tests.Exporters
+{
+    public class XmlSerializerTests
+    {
+        [Fact]
+        public void Serialize()
+        {
+            string expected =
+                "Start " +
+                "MockSource " +
+                    "IsValid True IsValid "+
+                    "IntNumber 42 IntNumber " +
+                    $"DoubleNumber {double.MaxValue.ToString(CultureInfo.InvariantCulture)} DoubleNumber " +
+                    $"LongNumber {long.MaxValue} LongNumber " +
+                    "Name Source Name " +
+                    "Items " +
+                        "Item Name i1 Name Item " +
+                        "Item Name i2 Name Item " +
+                    "Items " +
+                    "DoubleArray " +
+                        "Item 1 Item " +
+                        "Item 2 Item " +
+                    "DoubleArray " +
+                "MockSource " +
+                "End";
+            var source = new MockSource();
+            var writer = new MockXmlWriter();
+            IXmlSerializer serializer = new XmlSerializer(typeof(MockSource));
+
+            serializer.Serialize(writer, source);
+            string actual = writer.ToString();
+
+            Assert.Equal(expected, actual);
```

Comment:
@Teknikaali I think that we should leave it the way it is. I am very grateful that you wrote these tests.